lib/sender.js

```diff
@@ -64,70 +65,107 @@ module.exports = {
     @param {Array}   [options.criteria.categories] - a list of categories as strings
     @param {Array}   [options.criteria.variants] - a list of variantID's as strings
     @instance
+    @example
+    agSender(settings).then((client) => {
+        client.sender.send(message, options).then((response) => {
+            console.log('success', response);
+        })
+    });
 */
 function send(client) {
     return function send(message, options) {
-        return new Promise((resolve, reject) => {
-            options = options || {};
+        const body = composeSenderRequestBoddyForOneMessage(message, options);
+        return makeRequest(client, body);
+    };
+}
+
+/**
+    The sendBatch Method for multiple messages
+    @param {Array}  messages - An array of objects containing the fields described in send Method [{message, options}, ... ]
+    @instance
+    @example
+    agSender(settings).then((client) => {
+        client.sender.sendBatch(messages).then((response) => {
+            console.log('success', response);
+        })
+    });
+*/
+function sendBatch(client) {
+    return function sendBatch(messages) {
+        const body = messages.map(m => composeSenderRequestBoddyForOneMessage(m.message, m.options));
+        client.url += 'batch';
```

Comment:
In the end I found that the best solution is to add a new parameter "url" to avoid overriding the client. It could happen, in this particular case, that the url is messed up:
```
senderClient(settings)
            .then(client => {
                     client.sender.sendBatch(messages)
                     // All calls below will have the url already changed
                     client.sender.send(message, options)
                     ...
            });